lib/DXIL/DxilUtilDbgInfoAndMisc.cpp

```diff
@@ -130,23 +193,14 @@ bool MergeGepUse(Value *V) {
         // merge any GEP users of the untranslated bitcast
         addUsersToWorklist(V);
       }
+    } else if (isa<AddrSpaceCastOperator>(V)) {
+      addUsersToWorklist(V);
```

Comment:
I think generally collapsing addrspacecast chains would require an addition here to look at the pointer operand to see if it is also an addrspacecast.  If so, we can create a new merged addrspacecast, replace uses, and add it to the work list.

Reply:
Yes, that sounds like it would work. Not sure its worth the extra complexity in the code. Those should hopefully all be folded away by instcombine. If there was an hlsl source that could generate it that would be more compelling.